aom/src/main/java/com/nedap/archie/rules/ExpressionType.java

```diff
@@ -5,7 +5,7 @@
  * Created by pieter.bos on 27/10/15.
  */
 public enum ExpressionType {
-     BOOLEAN, STRING, INTEGER, REAL, DATE, TIME, DATETIME, DURATION;
+     BOOLEAN, STRING, INTEGER, REAL, DATE, TIME, DATETIME, DURATION, C_STRING;
```

Comment:
I am not sure if C_STRING should be added here. Expression type is used as the result of an expression when evaluating. 
I cannot think of an expression with result type C_STRING. Would that not be a boolean, whether it matches or not?

is there a reason in particular why this type would be needed here?

Reply:
This is the type of the ARCHETYPE_ID_CONSTRAINT atom, used with the matches operator. In general there can be a subtree whose construction is:
```
op = matches; type = Boolean
    opd_left = /path/to/some/value; type = <some primitive type>
    opd_right = C_XXX constraint object; type = <constraint for primitive type>
```
So for the specific case of matching archetype id, it is 
```
op = matches; type = Boolean
    opd_left = /archetype_id/value ; type = String
    opd_right = C_STRING regex to match archetype id; constraint object; type = C_STRING
```
As far as we can tell, the enum needs to have C_STRING in it, because without it, it doesn't allow any operand (or other node) to be a C_STRING. Most likely, all the C_XXX primitive types should be there, but we don't use them, so for now at least it won't matter. But the matches operator always has a sub-tree of the first form above, with a constraint matcher node on the right hand side.

Reply:
I guess that's correct - I guess we just don't always fill this. But then you also need all the other C_-types that can be expressed in the rules, at least C_TERMINOLOGY_CODE and C_BOOLEAN. Should not block these changes of course.